spec/services/update_subscription_address_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'spec_helper'
+
+describe UpdateSubscriptionAddress do
+
+  context "when updating a subscription" do
+    let(:user) { create(:user) }
+    let(:old_address) { create(:subscription_address, address1: "Old Address", user: user) }
+    let(:subscription) { create(:subscription, ship_address: old_address, bill_address: old_address, user: user) }
+
+    before do
+      @new_address = build_address_params(old_address)
+      @new_address[:address1] = "New Address"
+    end
```

Comment:
practically that does make more sense. i had started with the feature spec and once i had that fleshed out i moved to the service spec. i personally think it's overkill to test updating every single attribute on a subscription - if one attribute is successfully updated then the rest of them should update successfully as well. that's why since i had already written the feature spec and covered a lot of attributes there i didn't add more than one here. looking at it now though it i'm going to test updating more than one attribute it makes sense to include it in the unit test